Add ability to clone escorts for subsequent moves

dan-palmer opened the pull request:
Uses deep_cloneable to clone AR object graph.

Resets certain status fields on various tables so we know that
its an escort with sections that need reviewing.
- [x] clone implementation
- [x] basic spec
- [x] spec multiples(pending FG multiples additions #109)

dan-palmer commented on app/views/pdfs/_prisoner.html.erb:
```diff
@@ -2,47 +2,39 @@
-          <strong class="label"><%= t('.family_name') %></strong>
-          <div class="input"><%= prisoner.family_name %></div>
+        <span class="label"><%= t('.names') %></span>
+        <div><%= prisoner.family_name %> / <%= prisoner.forenames %></div>
```
This should be handled by the presenter me think

dan-palmer commented:
@gtvj I pulled this down & ran it locally - it looks a bit odd - think the layout is broken.

<img width="818" alt="screen shot 2016-04-14 at 17 53 17" src="https://cloud.githubusercontent.com/assets/1006365/14569126/95816d48-0334-11e6-880d-9bd80a8e6a81.png">

gtvj commented:
<img width="834" alt="screen shot 2016-04-18 at 12 38 32" src="https://cloud.githubusercontent.com/assets/16000203/14602945/b58195ca-0562-11e6-8763-1bc0264629e0.png">

dan-palmer commented:
@gtvj this is working fine now, thanks 👍

Commit:
Add ability to clone escorts for subsequent moves

Very much a WIP, uses deep_cloneable to clone AR object graph.

Resets certain status fields on various tables so we know that
its an escort with sections that need reviewing.

dan-palmer commented:
`deep_cloneable` causes a spec to fail:

``` ruby
1) Forms::CurrentOffence#save sets the data on the model
     Failure/Error: subject.save

     ActiveRecord::StatementInvalid:
       PG::NotNullViolation: ERROR:  null value in column "offences_id" violates not-null constraint
       DETAIL:  Failing row contains (d42466f5-2a85-4048-9164-9bd12e49efbd, null, Burglary, T123456, 2016-07-22 15:42:12.450868, 2016-07-22 15:42:12.450868).
       : INSERT INTO "current_offences" ("offence", "case_reference", "created_at", "updated_at") VALUES ($1, $2, $3, $4) RETURNING "id"
     # ./spec/forms/current_offence_spec.rb:22:in `block (3 levels) in <top (required)>'
```

🔬

Commit:
Fix exploding save spec

The addition of `deep_cloneable` causes the save spec to break.

The save specs are slightly odd though as they ultimately test
persistence, something in the domain of ActiveRecord. Its
questionable as to whether we should be speccing this most of
the time as Reform should own it.

Commit:
Update move attributes to purge on reuse

Following on from team discussion around re-use we've decided
to keep all the data around when cloning except the date of move.

Commit:
Add rough spec for cloning escort

Commit:
Include move destinations when cloning

Commit:
Reset escort models workflow status

Commit:
Spec multiples

& provide general helper method for ripping out the attributes
for comparison against the clones attributes

whoojemaflip committed:
Merge branch 'master' into clone-escort-for-reuse
